mitodl/ocw-studio
# added page type in deletable items

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5358

### Description (What does it do?)

This PR allows users to delete a WebsiteContent only if it is not referenced in other content.

### How can this be tested?
1. Make sure you have permission to delete content (Your email is added to the allowed list on PostHog)
2. Go to any resource and check if the Delete menu is there.
3. Refer to the instructions [here](https://github.com/mitodl/ocw-studio/pull/2322) for populating referencing content fields.
4. When the content is being referenced elsewhere, deletion is prevented and it shows a warning message like in the screenshot below

<img width="908" alt="Screenshot 2024-10-28 at 5 54 17 PM" src="https://github.com/user-attachments/assets/7343cd6a-f252-457a-bd3d-9eac5f5df706">

### Additional Testing
Feature Flag testing with PostHog
1. Ensure you set `POSTHOG_ENABLED=True`, `POSTHOG_API_HOST`, and `POSTHOG_PROJECT_API_KEY` values in `.env` in your local `ocw-studio` setup.
2. Rebuild containers
3. The flag should already enabled on the `dev` environment. If needed, visit PostHog to configure the `OCW_STUDIO_CONTENT_DELETABLE_REFERENCES` flag.
4. With the flag disabled, the content is deleted even if it has references
